lib/runtime/procs/reduce.js

```diff
@@ -152,8 +152,16 @@ class reduce_base extends periodic_fanin {
                 var row = that.groups.get_row(keyID);
                 var fns = row.fns;
                 var pt = { };
-                // new points get the batch end timestamp
-                if (eob) { pt.time = eob; }
+                // new points get the batch end timestamp...
+                if (eob) {
+                    pt.time = eob;
+                }
+                // ...and interval info from marks coming from batch,
+                // every of `reduce -every`, or undefined for unbatched
```

Comment:
The comment is slightly misleading — in case of unbatched stream, the interval isn’t set to `undefined` but not set at all. This is especially confusing because `undefined` is not a valid Juttle value.